apps/rush-lib/src/api/CommandLineConfiguration.ts

```diff
@@ -29,270 +96,399 @@ export class CommandLineConfiguration {
-    name: RushConstants.buildCommandName,
-    summary: "Build all projects that haven't been built, or have changed since they were last built.",
-    description:
-      'This command is similar to "rush rebuild", except that "rush build" performs' +
-      ' an incremental build. In other words, it only builds projects whose source files have changed' +
-      ' since the last successful build. The analysis requires a Git working tree, and only considers' +
-      ' source files that are tracked by Git and whose path is under the project folder. (For more details' +
-      ' about this algorithm, see the documentation for the "package-deps-hash" NPM package.) The incremental' +
-      ' build state is tracked in a per-project folder called ".rush/temp" which should NOT be added to Git. The' +
-      ' build command is tracked by the "arguments" field in the "package-deps_build.json" file contained' +
-      ' therein; a full rebuild is forced whenever the command has changed (e.g. "--production" or not).',
-    enableParallelism: true,
-    ignoreMissingScript: false,
-    ignoreDependencyOrder: false,
-    incremental: true,
-    allowWarningsInSuccessfulBuild: false,
-    safeForSimultaneousRushProcesses: false
-  };
-
-  public static readonly defaultRebuildCommandJson: CommandJson = {
-    commandKind: RushConstants.bulkCommandKind,
-    name: RushConstants.rebuildCommandName,
-    summary: 'Clean and rebuild the entire set of projects',
-    description:
-      'This command assumes that the package.json file for each project contains' +
-      ' a "scripts" entry for "npm run build" that performs a full clean build.' +
-      ' Rush invokes this script to build each project that is registered in rush.json.' +
-      ' Projects are built in parallel where possible, but always respecting the dependency' +
-      ' graph for locally linked projects.  The number of simultaneous processes will be' +
-      ' based on the number of machine cores unless overridden by the --parallelism flag.' +
-      ' (For an incremental build, see "rush build" instead of "rush rebuild".)',
-    enableParallelism: true,
-    ignoreMissingScript: false,
-    ignoreDependencyOrder: false,
-    incremental: false,
-    allowWarningsInSuccessfulBuild: false,
-    safeForSimultaneousRushProcesses: false
-  };
+  private _additionalPathFolders: string[] = [];
 
   /**
    * Use CommandLineConfiguration.loadFromFile()
    *
    * @internal
    */
   public constructor(commandLineJson: ICommandLineJson | undefined) {
-    if (commandLineJson) {
-      if (commandLineJson.phases) {
-        for (const phase of commandLineJson.phases) {
-          if (this.phases.has(phase.name)) {
-            throw new Error(
-              `In ${RushConstants.commandLineFilename}, the phase "${phase.name}" is specified ` +
-                'more than once.'
-            );
-          }
+    const commandsForPhase: Map<string, Set<IPhasedCommand>> = new Map();
```

Comment:
What is the `string` key?  One way to make this more obvious is to give the map a name like `commandsByPhaseName`, or `phaseCliquesByPhaseName`.